### DISCUSSION TITLE
Comparison between this and JavaScript or Wasm based solutions?

### DISCUSSION THREAD

**Discussion by fzyzcjy** [ethanblake4/dart_eval#70] [General]
(I am not against you! Just because if we are going to make it the solution, we have to think about all possible opposite voices by ourselves.)

So, what if someone (or us?) implements hot update via:

1. JavaScript?
2. Wasm?

(Hixie also mentioned these in 14330)

My non-mature answer:

* JS/Wasm have a much faster runtime (compared with dart_eval)
* But the bridge may be slow (or slower? or faster?)
* If bridge takes most of the time (we must have data to support this), then they do not have overall advantage

P.S. For some existing hot update solutions, here is a comparison article: https://juejin.cn/post/7033708048321347615 (Not in English, may need a translator...)

**Comment by ethanblake4** [comment]
Well, the main issue with Wasm is that you cannot compile Dart code to Wasm (yet). Even once you can compile Dart code to a Wasm _app_ I'm not sure you will be able to compile it to a Wasm _module_ (for use inside another app). Basically, all of the same caveats apply to Wasm as to a native Dart interpreter (eg compiler still has to be able to add bridge boundaries)

Javascript has been done by Hydro SDK and maybe others. It works well but I think most Flutter users don't want to use JS

**Comment by fzyzcjy** [comment]
> most Flutter users don't want to use JS

Yes, at least me :)

I was thinking about the dart2js official compiler though.
